src/Extension/FluentFilteredExtension.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace TractorCow\Fluent\Extension;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldAddNewButton;
+use SilverStripe\Forms\GridField\GridFieldConfig_RelationEditor;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataQuery;
+use SilverStripe\ORM\Queries\SQLSelect;
+use TractorCow\Fluent\Model\Locale;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Class FluentFilteredExtension
+ * @package TractorCow\Fluent\Extension
+ *
+ * @property FluentSiteTreeExtension|DataObject $owner
+ * @method Locale[] FilteredLocales()
+ */
+class FluentFilteredExtension extends DataExtension
+{
+    private static $many_many = [
+        'FilteredLocales' => Locale::class,
+    ];
+
+    public function updateSettingsFields(FieldList $fields)
+    {
+        $locales = Locale::get();
+
+        // If there are no Locales, then we're not adding any fields.
+        if ($locales->count() === 0) {
+            return;
+        }
+
+        $config = GridFieldConfig_RelationEditor::create();
+        $config->removeComponentsByType($config->getComponentByType(GridFieldAddNewButton::class));
+
+        // create a Checkbox group based on the array
+        $fields->addFieldToTab(
+            'Root.Settings',
+            GridField::create(
+                'FilteredLocales',
+                'Display in the following Locales',
+                $this->owner->FilteredLocales(),
+                $config
+            )
+        );
+    }
+
+    public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null)
+    {
+        // We don't want this logic applied in the CMS.
+        if (!FluentState::singleton()->getIsFrontend()) {
+            return;
+        }
+
+        $locale = $this->getDataQueryLocale($dataQuery);
+        if (!$locale) {
+            return;
+        }
+
+        $schema = DataObject::getSchema();
+        $table = $schema->baseDataTable(get_class($this->owner));
+        $filteredLocalesTable = $table . '_FilteredLocales';
+
+        $query->addInnerJoin(
+            $filteredLocalesTable,
+            "\"{$table}ID\" = \"{$table}\".\"ID\" AND \"{$locale->baseTable()}ID\" = ?",
+            null,
+            20,
+            [
+                $locale->ID,
+            ]
+        );
+    }
+
+    /**
+     * Get current locale from given dataquery
+     *
+     * @param DataQuery $dataQuery
+     * @return Locale|null
+     */
+    protected function getDataQueryLocale(DataQuery $dataQuery = null)
```

Comment:
I was unsure what to do about this. This method is a direct copy of `FluentExtension:: getDataQueryLocale()`, but this extension doesn't extend that (which, it probably shouldn't anyway).

Reply:
It's not too bad in terms of duplication. You could put it into a shared trait, maybe? I'd be inclined to live with it existing twice personally

Reply:
Yep a little duplication is fine; Otherwise you end up with tighter coupling.